### PR TITLE
[search] Filter features by locality type before GetTopLocalities.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: search/geocoder.cpp [hunk 1]
```diff
@@ -635,8 +637,8 @@ void Geocoder::FillLocalitiesTable(BaseContext const & ctx)
 {
   vector<Locality> preLocalities;
 
-  CBV filter;
-  filter.SetFull();
+  CBV filter = m_localitiesCache.Get(*m_context);
+  ;
```
Comment (mpimenov) [inline search/geocoder.cpp]: лишнее

File: search/geocoder.hpp [hunk 2]
```diff
@@ -148,7 +148,7 @@ class Geocoder
   void GoImpl(vector<shared_ptr<MwmInfo>> & infos, bool inViewport);
 
   template <typename Locality>
-  using LocalitiesCache = map<TokenRange, vector<Locality>>;
+  using TokenToLocalitiesCache = map<TokenRange, vector<Locality>>;
```
Comment (mpimenov) [inline search/geocoder.hpp]: Может, просто TokenToLocalities?

File: search/geocoder.cpp (hunk 1 above)
Comment (tatiana-yan) [reply search/geocoder.cpp]: fixed

File: search/geocoder.hpp (hunk 2 above)
Comment (tatiana-yan) [reply search/geocoder.hpp]: done

File: search/geocoder.hpp [hunk 3]
```diff
@@ -276,12 +277,12 @@ class Geocoder
...
   // and small villages and hamlets that are not.
-  LocalitiesCache<City> m_cities;
-  LocalitiesCache<Region> m_regions[Region::TYPE_COUNT];
+  TokenToLocalities<City> m_cities;
+  TokenToLocalities<Region> m_regions[Region::TYPE_COUNT];
   CitiesBoundariesTable const & m_citiesBoundaries;
 
   // Caches of features in rects. These caches are separated from
-  // TLocalitiesCache because the latter are quite lightweight and not
+  // TokenToLocalitiesCache because the latter are quite lightweight and not
```
Comment (mpimenov) [inline search/geocoder.hpp]: TokenToLocalities

Comment (tatiana-yan) [reply search/geocoder.hpp]: done

**Pull request merged**